omnigibson/robots/behaviorbot.py

```diff
@@ -0,0 +1,677 @@
+                "control_limits": self.control_limits,
+                "dof_idx": self.arm_control_idx[arm],
+                "command_input_limits": None,
+                "use_delta_commands": False,
+            }
+        return dic
+    
+    @property
+    def _default_gripper_multi_finger_controller_configs(self):
+        dic = {}
+        for arm in self.arm_names:
+            dic[arm] = {
+                "name": "MultiFingerGripperController",
+                "control_freq": self._control_freq,
+                "motor_type": "position",
+                "control_limits": self.control_limits,
+                "dof_idx": self.gripper_control_idx[arm],
+                "command_input_limits": None,
+                "mode": "independent",
+            }
+        return dic
+    
+    @property
+    def _default_camera_joint_controller_config(self):
+        return {
+            "name": "JointController",
+            "control_freq": self._control_freq,
+            "motor_type": "position",
+            "control_limits": self.control_limits,
+            "dof_idx": self.camera_control_idx,
+            "command_input_limits": None,
+            "use_delta_commands": False,
+        }
+
+    @property
+    def _default_controller_config(self):
+        controllers = {
+            "base": {"JointController": self._default_base_joint_controller_config},
+            "camera": {"JointController": self._default_camera_joint_controller_config},
+        }
+        controllers.update(
+            {
+                f"arm_{arm_name}": {"JointController": self._default_arm_joint_controller_configs[arm_name]}
+                for arm_name in self.arm_names
+            }
+        )
+        controllers.update(
+            {
+                f"gripper_{arm_name}": {
+                    "MultiFingerGripperController": self._default_gripper_multi_finger_controller_configs[arm_name],
+                    "JointController": self._default_gripper_joint_controller_configs[arm_name],
+                }
+                for arm_name in self.arm_names
+            }
+        )
+        return controllers
+
+    def load(self):
+        prim = super(Behaviorbot, self).load()
+        for part in self.parts.values():
+            part.load()
+        return prim
+
+    def _post_load(self):
+        super()._post_load()
+
+    def _create_discrete_action_space(self):
+        raise ValueError("Behaviorbot does not support discrete actions!")
+    
+    def update_controller_mode(self):
+        super().update_controller_mode()
+        # overwrite robot params (e.g. damping, stiffess, max_effort) here
+        for link in self.links.values():
+            link.ccd_enabled = True
+        for arm in self.arm_names:
+            for finger_link in self.finger_links[arm]:
+                finger_link.mass = 0.02
+        self.links["base"].mass = 15
+        self.links["lh_palm"].mass = 0.3
+        self.links["rh_palm"].mass = 0.3
```

Comment:
hmm these things don't really belong to controller (as in they shouldn't be in the function `update_controller_mode`?

Reply:
Yes I'll remove these